test/components/account/login.spec.js

```diff
@@ -233,4 +243,175 @@ describe('AccountLogin', () => {
+      const result = await app.vm.$router.push('/not-found');
+      isNavigationFailure(result, NavigationFailureType.aborted).should.be.true();
+    });
+  });
+
+  describe('next query parameter if OIDC is enabled', () => {
+    it('appends ?next to the link', async () => {
+      const component = await load('/login?next=%2Fusers', {
+        container: oidcContainer,
+        root: false
+      });
+      const href = component.get('a').attributes('href');
+      href.should.equal('/v1/oidc/login?next=%2Fusers');
+    });
+
+    it('does not append ?next if it was specified twice', async () => {
+      const component = await load('/login?next=%2Fusers&next=%2Faccount%2Fedit', {
+        container: oidcContainer,
+        root: false
+      });
+      component.get('a').attributes('href').should.equal('/v1/oidc/login');
+    });
+
+    it('does not append ?next if it has no value', async () => {
+      const component = await load('/login?next', {
+        container: oidcContainer,
+        root: false
+      });
+      component.get('a').attributes('href').should.equal('/v1/oidc/login');
+    });
+  });
+
+  describe('OIDC error', () => {
+    it('shows an alert for auth-ok-user-not-found', async () => {
+      const component = await load('/login?oidcError=auth-ok-user-not-found', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('There is no Central account associated with your email address.');
+      });
+    });
+
+    it('shows an alert for email-claim-not-provided', async () => {
+      const component = await load('/login?oidcError=email-claim-not-provided', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('Central could not access the email address associated with your account.');
+      });
+    });
+
+    it('shows an alert for email-not-verified', async () => {
+      const component = await load('/login?oidcError=email-not-verified', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('Your email address has not been verified by your login server.');
+      });
+    });
+
+    it('does not show an alert if there are two errors', async () => {
+      const component = await load('/login?oidcError=auth-ok-user-not-found&oidcError=email-claim-not-provided', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.not.alert();
+    });
+
+    it('does not show an alert if query parameter has no value', async () => {
+      const component = await load('/login?oidcError', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.not.alert();
+    });
+
+    it('does not show an alert for an error whose name is invalid', async () => {
```

Comment:
how about showing generic error message?

Reply:
We don't actually really have a generic error message. We could add one, but it's something that would need to be translated. This test is mostly here just to verify that nothing breaks in this case. I don't expect that users will encounter this case in normal usage.